In [86]:
#pip3 install python-twitter
import twitter
import pandas as pd
import json

In [2]:
api = twitter.Api(consumer_key='RNbWRckUdr1mzMYe2ir2S44aP',
consumer_secret='FVB0RX0nGo1VMRAMF603g7jOKuZdKfCLo0SjXWv2T15wHvLMRy',
access_token_key='1220749681900191744-JDImKJjEF4NadKsiKekx82zDI9FyQW',
access_token_secret='bQaUWESSjNkqhO3d2Sc4pLPmNDYLr5DiRIIS5ko5Sg4yv')

In [3]:
print(api.VerifyCredentials())

{"created_at": "Fri Jan 24 16:46:29 +0000 2020", "default_profile": true, "friends_count": 7, "id": 1220749681900191744, "id_str": "1220749681900191744", "name": "Luis Felipe Cubero", "profile_background_color": "F5F8FA", "profile_image_url": "http://pbs.twimg.com/profile_images/1220750095768903680/FxWNsUy0_normal.jpg", "profile_image_url_https": "https://pbs.twimg.com/profile_images/1220750095768903680/FxWNsUy0_normal.jpg", "profile_link_color": "1DA1F2", "profile_sidebar_border_color": "C0DEED", "profile_sidebar_fill_color": "DDEEF6", "profile_text_color": "333333", "profile_use_background_image": true, "screen_name": "LuisFelipeCube1"}


In [4]:
statuses = api.GetUserTimeline(screen_name='realdonaldtrump')
statuses

[Status(ID=1223038027234267137, ScreenName=realDonaldTrump, Created=Fri Jan 31 00:19:23 +0000 2020, Text='Great poll in Iowa, where I just landed for a Big Rally! #KAG2020🇺🇸 https://t.co/4YCo01XYCn')]

In [35]:
api.GetSearch(term='#neuralink', since=2020-1-1, count=100)

[Status(ID=1223079561765761025, ScreenName=TrkFreak, Created=Fri Jan 31 03:04:25 +0000 2020, Text='RT @HogDexter: Elon Musk Quotes... 💫https://t.co/QMD6cr7sCe💫 #elonmusk #tesla #spacex #boringco #starlink #neuralink https://t.co/ggcBsTjRNr'),
 Status(ID=1223015865588817920, ScreenName=HogDexter, Created=Thu Jan 30 22:51:19 +0000 2020, Text='RT @HogDexter: Elon Musk Quotes... 💫https://t.co/QMD6cr7sCe💫 #elonmusk #tesla #spacex #boringco #starlink #neuralink https://t.co/ggcBsTjRNr'),
 Status(ID=1223011342279479296, ScreenName=symshark, Created=Thu Jan 30 22:33:21 +0000 2020, Text='“And can you offer me proof of your existence? How can you, when neither modern science nor philosophy can explain… https://t.co/ncWSZqGd06'),
 Status(ID=1222948263210430466, ScreenName=coriandreas, Created=Thu Jan 30 18:22:41 +0000 2020, Text='@ZDFheute Es gibt zwei Möglichkeiten: Wir blockieren neue Technologien und jammern weiter... Oder wir entscheiden uns für #Neuralink.'),
 Status(ID=1222947919445286912, 

In [6]:
user = api.GetUser(screen_name='delfinomachin')

In [7]:
user.__dict__.keys()

dict_keys(['param_defaults', 'contributors_enabled', 'created_at', 'default_profile', 'default_profile_image', 'description', 'email', 'favourites_count', 'followers_count', 'following', 'friends_count', 'geo_enabled', 'id', 'id_str', 'lang', 'listed_count', 'location', 'name', 'notifications', 'profile_background_color', 'profile_background_image_url', 'profile_background_image_url_https', 'profile_background_tile', 'profile_banner_url', 'profile_image_url', 'profile_image_url_https', 'profile_link_color', 'profile_sidebar_border_color', 'profile_sidebar_fill_color', 'profile_text_color', 'profile_use_background_image', 'protected', 'screen_name', 'status', 'statuses_count', 'time_zone', 'url', 'utc_offset', 'verified', 'withheld_in_countries', 'withheld_scope', '_json'])

In [8]:
user._json

{'id': 1006173880568942593,
 'id_str': '1006173880568942593',
 'name': 'Diego Delfino',
 'screen_name': 'delfinomachin',
 'location': 'San José, Costa Rica',
 'profile_location': None,
 'description': 'Contacto: diego@delfino.cr',
 'url': 'https://t.co/7p5Ok7995u',
 'entities': {'url': {'urls': [{'url': 'https://t.co/7p5Ok7995u',
     'expanded_url': 'http://Delfino.CR',
     'display_url': 'Delfino.CR',
     'indices': [0, 23]}]},
  'description': {'urls': []}},
 'protected': False,
 'followers_count': 5174,
 'friends_count': 699,
 'listed_count': 6,
 'created_at': 'Mon Jun 11 13:58:41 +0000 2018',
 'favourites_count': 3688,
 'utc_offset': None,
 'time_zone': None,
 'geo_enabled': True,
 'verified': False,
 'statuses_count': 4819,
 'lang': None,
 'status': {'created_at': 'Thu Jan 30 19:09:29 +0000 2020',
  'id': 1222960038173724672,
  'id_str': '1222960038173724672',
  'text': '@Guiller12790376 🤣 Solo vos. 🖤',
  'truncated': False,
  'entities': {'hashtags': [],
   'symbols': [],
   '

In [9]:
api.GetStatus(1222971565844979714)._json

{'created_at': 'Thu Jan 30 19:55:17 +0000 2020',
 'id': 1222971565844979714,
 'id_str': '1222971565844979714',
 'text': '@julio https://t.co/biKuktveVr',
 'truncated': False,
 'entities': {'hashtags': [],
  'symbols': [],
  'user_mentions': [{'screen_name': 'Julio',
    'name': 'Julio Contreras',
    'id': 12729802,
    'id_str': '12729802',
    'indices': [0, 6]}],
  'urls': [{'url': 'https://t.co/biKuktveVr',
    'expanded_url': 'https://twitter.com/elifeinzaig/status/1222968352089563136',
    'display_url': 'twitter.com/elifeinzaig/st…',
    'indices': [7, 30]}]},
 'source': '<a href="http://twitter.com/download/iphone" rel="nofollow">Twitter for iPhone</a>',
 'in_reply_to_status_id': None,
 'in_reply_to_status_id_str': None,
 'in_reply_to_user_id': 12729802,
 'in_reply_to_user_id_str': '12729802',
 'in_reply_to_screen_name': 'Julio',
 'user': {'id': 2891285033,
  'id_str': '2891285033',
  'name': 'Esteban Araya Oviedo',
  'screen_name': 'EarayaOviedo',
  'location': 'Flores, Heredi

In [40]:
test= api.GetSearch(since=2020-1-1, count=100, geocode="9.93333,-84.08333,5km")
test

[Status(ID=1223085148985548801, ScreenName=MariiPaz0704, Created=Fri Jan 31 03:26:37 +0000 2020, Text='Vi esto por si alguien ocupa trabajo https://t.co/3BOxyzl0b8'),
 Status(ID=1223083751481233408, ScreenName=SquatchTheWorld, Created=Fri Jan 31 03:21:04 +0000 2020, Text='@rmcapocaccia @MelissaSweazy Yeah I saw her do the reading a few years ago at the NCRM. Amazing woman'),
 Status(ID=1223083279085117440, ScreenName=richdeman, Created=Fri Jan 31 03:19:12 +0000 2020, Text='Daño irreversible: hallan agua tibia debajo del glaciar ‘’más peligroso del mundo’’ https://t.co/0TSGdd3UAW'),
 Status(ID=1223082575822053377, ScreenName=xluiskb, Created=Fri Jan 31 03:16:24 +0000 2020, Text='La mae que va al lado mío acaba de abrir un Jet y me saboree la pacha 😂'),
 Status(ID=1223082514895470594, ScreenName=SquatchTheWorld, Created=Fri Jan 31 03:16:09 +0000 2020, Text='Mike Pence is a fool https://t.co/iBGUM8dFIi'),
 Status(ID=1223082454006861825, ScreenName=xluiskb, Created=Fri Jan 31 03:15:55 +000

In [26]:
test[0]

Status(ID=1223074372484845568, ScreenName=R4K3lobo, Created=Fri Jan 31 02:43:48 +0000 2020, Text='https://t.co/zb3m7f8PkH')

In [25]:
test[0].__dict__.keys()

dict_keys(['param_defaults', 'contributors', 'coordinates', 'created_at', 'current_user_retweet', 'favorite_count', 'favorited', 'full_text', 'geo', 'hashtags', 'id', 'id_str', 'in_reply_to_screen_name', 'in_reply_to_status_id', 'in_reply_to_user_id', 'lang', 'location', 'media', 'place', 'possibly_sensitive', 'quoted_status', 'quoted_status_id', 'quoted_status_id_str', 'retweet_count', 'retweeted', 'retweeted_status', 'scopes', 'source', 'text', 'truncated', 'urls', 'user', 'user_mentions', 'withheld_copyright', 'withheld_in_countries', 'withheld_scope', 'tweet_mode', '_json'])

In [36]:
print (test[0].id)
print (test[0].user.screen_name)

1223074372484845568
R4K3lobo


In [30]:
test[0]._json

{'created_at': 'Fri Jan 31 02:43:48 +0000 2020',
 'id': 1223074372484845568,
 'id_str': '1223074372484845568',
 'text': 'https://t.co/zb3m7f8PkH',
 'truncated': False,
 'entities': {'hashtags': [],
  'symbols': [],
  'user_mentions': [],
  'urls': [],
  'media': [{'id': 1223074361919393793,
    'id_str': '1223074361919393793',
    'indices': [0, 23],
    'media_url': 'http://pbs.twimg.com/media/EPk7puwX0AEhF1R.jpg',
    'media_url_https': 'https://pbs.twimg.com/media/EPk7puwX0AEhF1R.jpg',
    'url': 'https://t.co/zb3m7f8PkH',
    'display_url': 'pic.twitter.com/zb3m7f8PkH',
    'expanded_url': 'https://twitter.com/R4K3lobo/status/1223074372484845568/photo/1',
    'type': 'photo',
    'sizes': {'thumb': {'w': 150, 'h': 150, 'resize': 'crop'},
     'small': {'w': 357, 'h': 680, 'resize': 'fit'},
     'medium': {'w': 629, 'h': 1200, 'resize': 'fit'},
     'large': {'w': 1074, 'h': 2048, 'resize': 'fit'}}}]},
 'extended_entities': {'media': [{'id': 1223074361919393793,
    'id_str': '12230

In [32]:
test[0].user.screen_name

'R4K3lobo'

In [43]:
test[0].user.__dict__.keys()

dict_keys(['param_defaults', 'contributors_enabled', 'created_at', 'default_profile', 'default_profile_image', 'description', 'email', 'favourites_count', 'followers_count', 'following', 'friends_count', 'geo_enabled', 'id', 'id_str', 'lang', 'listed_count', 'location', 'name', 'notifications', 'profile_background_color', 'profile_background_image_url', 'profile_background_image_url_https', 'profile_background_tile', 'profile_banner_url', 'profile_image_url', 'profile_image_url_https', 'profile_link_color', 'profile_sidebar_border_color', 'profile_sidebar_fill_color', 'profile_text_color', 'profile_use_background_image', 'protected', 'screen_name', 'status', 'statuses_count', 'time_zone', 'url', 'utc_offset', 'verified', 'withheld_in_countries', 'withheld_scope', '_json'])

In [34]:
test[0].user.name

'Fanta Kolita :3'

Status(ID=1223079561765761025, ScreenName=TrkFreak, Created=Fri Jan 31 03:04:25 +0000 2020, Text='RT @HogDexter: Elon Musk Quotes... 💫https://t.co/QMD6cr7sCe💫 #elonmusk #tesla #spacex #boringco #starlink #neuralink https://t.co/ggcBsTjRNr')

tabla:
ID_tweet:
user_screen_namne:
created_date:
hora:
text:
user_info
name
location
description
followers_count
friends_count
user_created_at
count_twitter:
retweet_count:
favorite_count


In [84]:
print (test[0].id)
print (test[0].user.screen_name)
# print (test[0].created_at)
# print (type(test[0].created_at))
print (test[0].created_at[4:11]+test[0].created_at[-4:])
print(test[0].created_at[12:20])
print(test[0].text)
print(test[0].user.name)
print(test[0].user.location)
print(test[0].user.description)
print(test[0].user.followers_count)
print(test[0].user.friends_count)
print (test[0].user.created_at[4:11]+test[0].created_at[-4:])
print (test[0].user.created_at[4:11]+test[0].created_at[-4:])
print(test[0].user.statuses_count)

print(test[0].retweet_count)
print(test[0].favorite_count)


1223085148985548801
MariiPaz0704
Jan 31 2020
3:26:37 
Vi esto por si alguien ocupa trabajo https://t.co/3BOxyzl0b8
Maria Paz GV
Costa Rica
snap:pazi_0704
soy de coro
adoro cocinar
91
241
Aug 30 2020
Aug 30 2020
537
0
0


In [68]:
test[0].status

AttributeError: 'Status' object has no attribute 'status'

In [54]:
test[0].user._json

{'id': 70022478,
 'id_str': '70022478',
 'name': 'Maria Paz GV',
 'screen_name': 'MariiPaz0704',
 'location': 'Costa Rica',
 'description': 'snap:pazi_0704\nsoy de coro\nadoro cocinar',
 'url': None,
 'entities': {'description': {'urls': []}},
 'protected': False,
 'followers_count': 91,
 'friends_count': 241,
 'listed_count': 0,
 'created_at': 'Sun Aug 30 02:09:20 +0000 2009',
 'favourites_count': 1227,
 'utc_offset': None,
 'time_zone': None,
 'geo_enabled': True,
 'verified': False,
 'statuses_count': 537,
 'lang': None,
 'contributors_enabled': False,
 'is_translator': False,
 'is_translation_enabled': False,
 'profile_background_color': '8C8B91',
 'profile_background_image_url': 'http://abs.twimg.com/images/themes/theme10/bg.gif',
 'profile_background_image_url_https': 'https://abs.twimg.com/images/themes/theme10/bg.gif',
 'profile_background_tile': True,
 'profile_image_url': 'http://pbs.twimg.com/profile_images/1192304090472177665/POZul_ko_normal.jpg',
 'profile_image_url_https'

In [62]:
api.GetRetweeters(1223128794396938241, cursor=None, count=100, stringify_ids=False)

[855950603675217924,
 969343815889969154,
 570561007,
 768433682,
 1181623650740576256,
 2993467498,
 3076048140,
 2915722919,
 4675683350,
 863174993185779714,
 53922474,
 964496207761551360,
 2882916733,
 1106946453375774720,
 2417759161,
 4579369779,
 2902314918,
 24213525,
 1610732888,
 1065355220329398272,
 27766981,
 2561989309,
 3644683453,
 1186402298035085313,
 723547824592437248,
 359163895,
 3437321536,
 55220919,
 925185460330291200,
 2872477975,
 938429272812900352,
 1896815268,
 2337427615,
 3322924514,
 1218999072457023488,
 182063696,
 179496511,
 1129659627925319682,
 711950597658902528,
 858808212530708480,
 942807572,
 2574135345,
 3188580347,
 895051989826105344,
 764849648028774400,
 2944508276,
 497368201,
 853840799414484992,
 276065914,
 1206788489905344513,
 466891902,
 2175940251,
 1217122892112125956,
 273234285,
 716264026800427008,
 748995670367952896,
 1178689691752783872,
 934996759935365121,
 575341554,
 1039610164779855872,
 455080733,
 9056309687776706

In [79]:
api.GetSearch(1223038027234267137)[0].__dict__.keys()

dict_keys(['param_defaults', 'contributors', 'coordinates', 'created_at', 'current_user_retweet', 'favorite_count', 'favorited', 'full_text', 'geo', 'hashtags', 'id', 'id_str', 'in_reply_to_screen_name', 'in_reply_to_status_id', 'in_reply_to_user_id', 'lang', 'location', 'media', 'place', 'possibly_sensitive', 'quoted_status', 'quoted_status_id', 'quoted_status_id_str', 'retweet_count', 'retweeted', 'retweeted_status', 'scopes', 'source', 'text', 'truncated', 'urls', 'user', 'user_mentions', 'withheld_copyright', 'withheld_in_countries', 'withheld_scope', 'tweet_mode', '_json'])

In [94]:
api.GetSearch(1223128794396938241)[0]._json

{'created_at': 'Fri Jan 31 06:34:48 +0000 2020',
 'id': 1223132503185248257,
 'id_str': '1223132503185248257',
 'text': 'Up at 1:30am listening to an Elon Musk original on soundcloud. What happened to me?? https://t.co/6k0oYMjWVK',
 'truncated': False,
 'entities': {'hashtags': [],
  'symbols': [],
  'user_mentions': [],
  'urls': [{'url': 'https://t.co/6k0oYMjWVK',
    'expanded_url': 'https://twitter.com/elonmusk/status/1223128794396938241',
    'display_url': 'twitter.com/elonmusk/statu…',
    'indices': [85, 108]}]},
 'metadata': {'result_type': 'popular', 'iso_language_code': 'en'},
 'source': '<a href="https://mobile.twitter.com" rel="nofollow">Twitter Web App</a>',
 'in_reply_to_status_id': None,
 'in_reply_to_status_id_str': None,
 'in_reply_to_user_id': None,
 'in_reply_to_user_id_str': None,
 'in_reply_to_screen_name': None,
 'user': {'id': 220484183,
  'id_str': '220484183',
  'name': 'Cristian Nyari',
  'screen_name': 'Cnyari',
  'location': 'New York City',
  'description'

In [83]:
api.GetSearch(1223038027234267137)[0]._json

{'created_at': 'Fri Jan 31 00:19:23 +0000 2020',
 'id': 1223038027234267137,
 'id_str': '1223038027234267137',
 'text': 'Great poll in Iowa, where I just landed for a Big Rally! #KAG2020🇺🇸 https://t.co/4YCo01XYCn',
 'truncated': False,
 'entities': {'hashtags': [{'text': 'KAG2020', 'indices': [57, 65]}],
  'symbols': [],
  'user_mentions': [],
  'urls': [],
  'media': [{'id': 1223038022867935232,
    'id_str': '1223038022867935232',
    'indices': [68, 91],
    'media_url': 'http://pbs.twimg.com/media/EPkamhPW4AAP_lP.jpg',
    'media_url_https': 'https://pbs.twimg.com/media/EPkamhPW4AAP_lP.jpg',
    'url': 'https://t.co/4YCo01XYCn',
    'display_url': 'pic.twitter.com/4YCo01XYCn',
    'expanded_url': 'https://twitter.com/realDonaldTrump/status/1223038027234267137/photo/1',
    'type': 'photo',
    'sizes': {'thumb': {'w': 150, 'h': 150, 'resize': 'crop'},
     'small': {'w': 680, 'h': 507, 'resize': 'fit'},
     'medium': {'w': 1129, 'h': 841, 'resize': 'fit'},
     'large': {'w': 1129

In [93]:
api.GetSearch(1223038027234267137)[0]._json['entities']['hashtags'][0]['text']

'KAG2020'